### PR TITLE
fix: fixed type hints

### PR DESCRIPTION
**Pull request by zeqri**
Reference Issues/PRs

removes warnings when running  ```mkdocs build --clean --strict```

### CODE REVIEW

File: whittle/training_strategies/ats.py
Comment (timurcarstensen) [inline whittle/training_strategies/ats.py:1]: imports are supposed to be ordered in this format: 
1. standard library imports (e.g., `os`, `typing`, etc. see [here](https://docs.python.org/3/library/index.html))
2. third party imports (e.g., `torch`, `numpy`, etc.)
3. local imports (your own code such as the code in whittle)

File: whittle/training_strategies/random_linear.py
Comment (timurcarstensen) [inline whittle/training_strategies/random_linear.py:1]: see previous comment

File: whittle/training_strategies/sandwich.py
Comment (timurcarstensen) [inline whittle/training_strategies/sandwich.py:1]: see previous comment

**Review by timurcarstensen — CHANGES_REQUESTED** [review]
Thanks a lot! :) could you also make sure that you add update the flags for mkdocs in: 
- [x] `docs.yml`
- [ ] `release.yml`

File: whittle/training_strategies/ats.py
Comment (zeqri) [reply whittle/training_strategies/ats.py:1]: understood

**Pull request merged**